sky/backends/backend_utils.py

```diff
@@ -2082,10 +2084,31 @@ def run_ray_status_to_check_ray_cluster_healthy() -> bool:
                 try:
                     ready_head, ready_workers, output, stderr = (
                         get_node_counts_from_ray_status(head_runner))
-                except RuntimeError as e:
+                except exceptions.CommandError as e:
                     logger.debug(f'Refreshing status ({cluster_name!r}) attempt'
                                  f' {i}: {common_utils.format_exception(e)}')
                     if cloud_name != 'kubernetes':
+                        # Non-k8s clusters can be manually restarted and:
+                        # 1. Get new IP addresses, or
+                        # 2. Not have the SkyPilot runtime setup
+                        #
+                        # So we should surface a message to the user to
+                        # help them recover from this inconsistent state.
+                        if (_RAY_CLUSTER_NOT_FOUND_MESSAGE in e.error_msg or
+                            (e.detailed_reason is not None and
+                             _SSH_CONNECTION_TIMED_OUT_PATTERN.search(
+                                 e.detailed_reason.strip()) is not None)):
```

Comment:
```suggestion
                        # Non-k8s clusters can be manually restarted and:
                        # 1. Get new IP addresses, or
                        # 2. Not have the SkyPilot runtime setup
                        #
                        # So we should surface a message to the user to
                        # help them recover from this inconsistent state.
                        has_new_ip_addr = (e.detailed_reason is not None and
                             _SSH_CONNECTION_TIMED_OUT_PATTERN.search(
                                 e.detailed_reason.strip()) is not None)
                        runtime_not_setup = (_RAY_CLUSTER_NOT_FOUND_MESSAGE in e.error_msg)
                        if has_new_ip_addr or runtime_not_setup:
```
just so it's more clear what condition is checking for what